share/completions/doas.fish

```diff
@@ -0,0 +1,36 @@
+
+#
+# Completion for doas https://github.com/multiplexd/doas
+# based on the sudo completions
+#
+
+function __fish_doas_print_remaining_args
+    set -l tokens (commandline -opc) (commandline -ct)
+    set -e tokens[1]
+    # These are all the options mentioned in the man page for openbsd's "doas" (in that order).
+    set -l opts a= C= L n s u=
+    argparse -s $opts -- $tokens 2>/dev/null
+    # The remaining argv is the subcommand with all its options, which is what
+    # we want.
+    if test -n "$argv"
+        and not string match -qr '^-' $argv[1]
+        echo $argv
```

Comment:
This output isn't currently used, though I'd imagine you'll want to do so - see [the sudo completions](https://github.com/fish-shell/fish-shell/blob/master/share/completions/sudo.fish).

The reason that does the `echo` is because it then uses that for `complete -C` (`--do-complete=`), which is better than using the stock `__fish_complete_subcommand` function because it used `argparse` to remove the options.

As an illustration of this general problem, imagine you had

```fish
git -C log status
```

as your commandline. What is the command here? If you knew nothing about `git`, you'd say `log`, with `status` being an argument to it and `-C` being a boolean flag. But actually the command is `status`, because `log` is an argument to the `-C` option.

That's why argparse is useful, because it can solve those cases. You're almost there, but you need to do the last step.

Reply:
I'll have a closer look at this. This is my first time writing completions and I couldn't find much documentation about it, so this is based on other completions.